react-design/my-app/src/RotatingBanner.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from 'react';
+
+type Props = {
+  items: string[];
+};
+
+export function RotatingBanner({ items }: Props) {
+  const [currentItemIndex, setCurrentItemIndex] = useState(0);
+
+  const handlePrev = () => {
+    setCurrentItemIndex((prevIndex) =>
+      prevIndex === 0 ? items.length - 1 : prevIndex - 1
+    );
```

Comment:
This (and the one below) could be simplified a bit if you just use `currentItemIndex` instead of the function version of the setter:
`setCurrentItemIndex(currentItemIndex === 0 ? items.length - 1 : currentItemIndex - 1);`